### ISSUE TITLE
Automatically launch new DBServers when task is LOST

### ISSUE THREAD

**Issue by mikestaub** [arangodb/arangodb-mesos-framework#11]
If I launch the framework on DC/OS 1.9, then manually terminate the EC2 instance that one of the DBServers is running on, that task never comes back up.

That data is of course lost, as the storage for the instance is gone, but the framework should try its best to recover the cluster by launching a new DBServer task after some time. (15 minutes perhaps)

**Comment by mikestaub** [comment]
Even more importantly, it seems the DBServer task is not relaunched even when the underlying instance is simply rebooted. I thought that the framework is using persistent volumes, so it should be able to find the node that has the data and reconnect.

**Comment by neunhoef** [comment]
I am aware of one issue which might be related here and we are in the process of fixing this: If the same Mesos node runs an agent instance then the system tries to restart the agent first, which does not work, because its data is lost as well. Therefore it never gets to replacing the DBServer. Is it possible that this is what has happened in your case?
As I said, we are in the process of fixing this for the 3.2 release, such that the agent is replaced first and then the DBServer.

**Comment by mikestaub** [comment]
Yes that is what happened now that you mention it.

However, if I simply reboot the EC2 instance, that means the data (persistent mesos volume) should still be available to discover after the node comes back up, no? This is of course not true if the instance is terminated.

**Comment by neunhoef** [comment]
Right. If you only reboot, the persistent volumes should be reoffered to us by the new mesos-agent that is restarted, and then the framework picks them up again to restart "the same" arangod instance. This used to work, but I have not tested it myself with DC/OS 1.9. Also, I have sometimes seen that the cloud formation templates of DC/OS set up AWS instances with "ephemeral disks". I have myself lost data by "stopping" them, which means termination for AWS and destroys the volume. Can it be that this happens with a reboot as well?
Which version of DC/OS are you using and are you using the default cloud formation template or a custom one?

**Comment by mikestaub** [comment]
I am using DC/OS 1.9. We are indeed using ephemeral disk storage but as the docs describe, the data will persist on reboots (but not stopping or termination)
http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/InstanceStorage.html

You mentioned earlier that 3.1 will attempt to start the agent task first, but will fail because the data is lost. But how could the data have been lost if the instance was just rebooted, not stopped or terminated?

**Comment by neunhoef** [comment]
This is a mystery to me. I will need to try out the cloud formation template and see what happens at a reboot. I know that this has worked with earlier versions of DC/OS, but I have not seen it with my own eyes with DC/OS 1.9. You are right, a reboot must preserve the persistent volumes and our framework must pick them up and restart tasks and all must be well. If this does not work it is a rather serious bug.

**Comment by fceller** [comment]
@neunhoef what is the status here?

**Comment by neunhoef** [comment]
I do not know. Other things had higher prio so far. Need to try out on AWS.
